lightning/src/routing/router.rs

```diff
@@ -887,49 +967,51 @@ where L::Target: Logger {
-									// max_value means we'll always fail
-									// the old_entry.total_fee_msat > total_fee_msat check
-									None => total_fee_msat = u64::max_value(),
-									Some(fee_msat) => {
+								total_fee_msat = compute_fees(amount_to_transfer_over_msat, $candidate.fees())
+									.and_then(|fee_msat| {
 										hop_use_fee_msat = fee_msat;
 										total_fee_msat += hop_use_fee_msat;
-										// When calculating the lowest inbound fees to a node, we
-										// calculate fees here not based on the actual value we think
-										// will flow over this channel, but on the minimum value that
-										// we'll accept flowing over it. The minimum accepted value
-										// is a constant through each path collection run, ensuring
-										// consistent basis. Otherwise we may later find a
-										// different path to the source node that is more expensive,
-										// but which we consider to be cheaper because we are capacity
-										// constrained and the relative fee becomes lower.
-										match compute_fees(minimal_value_contribution_msat, old_entry.src_lowest_inbound_fees)
-												.map(|a| a.checked_add(total_fee_msat)) {
-											Some(Some(v)) => {
-												total_fee_msat = v;
-											},
-											_ => {
-												total_fee_msat = u64::max_value();
-											}
-										};
-									}
-								}
+										compute_fees(minimal_value_contribution_msat, old_entry.src_lowest_inbound_fees)
+									})
+									// When calculating the lowest inbound fees to a node, we
+									// calculate fees here not based on the actual value we think
+									// will flow over this channel, but on the minimum value that
+									// we'll accept flowing over it. The minimum accepted value
+									// is a constant through each path collection run, ensuring
+									// consistent basis. Otherwise we may later find a
+									// different path to the source node that is more expensive,
+									// but which we consider to be cheaper because we are capacity
+									// constrained and the relative fee becomes lower.
+									.and_then(|a| a.checked_add(total_fee_msat))
+									// max_value means we'll always fail
+									// the old_entry.total_fee_msat > total_fee_msat check
+									.unwrap_or_else(|| u64::max_value());
 							}
 
-							let path_penalty_msat = $next_hops_path_penalty_msat.checked_add(
-								scorer.channel_penalty_msat($chan_id.clone(), amount_to_transfer_over_msat, Some(*available_liquidity_msat),
-									&$src_node_id, &$dest_node_id)).unwrap_or_else(|| u64::max_value());
+							let effective_capacity_msat = $candidate.effective_capacity().as_msat();
+							let params = ChannelUseParameters {
+								amount_msat: amount_to_transfer_over_msat,
+								inflight_htlc_msat: effective_capacity_msat - available_liquidity_msat,
+								fees_msat: cmp::max(total_fee_msat, path_htlc_minimum_msat),
+								effective_capacity_msat,
```

Comment:
Presumably we want to include both the channel's value (if available) and htlc-max here.

Reply:
I had been thinking about passing `EffectiveCapacity` directly rather than `u64` to allow for a way to discern this, though I suppose if it is `EffectiveCapacity::Total` then we wouldn't get the HTLC max.

Another idea would be to pass `DirectedChannelInfo` instead, which includes `source` and `target`, to provide the "static" channel information. Then `ChannelUseParameters` could be limited to information for the current channel use (amount, fees, in-flight HTLCs). Probably would want to precompute `EffectiveCapacity` and `htlc_maximum_msat` when constructing the `DirectedChannelInfo`, I'd imagine.

Reply:
Yea, I think passing `DirectedChannelInfo` would make sense.